src/shared/prosemirror-plugins/image-upload.ts

```diff
@@ -116,7 +116,7 @@ export class ImageUploader implements PluginView {
         this.uploadField.className = "d-none";
         this.uploadField.accept = "image/*";
         this.uploadField.multiple = false;
-        this.uploadField.id = "fileUpload";
+        this.uploadField.id = "fileUpload" + (Math.random() * 10000).toFixed(0);
```

Comment:
This looks good for now. We have a concept of an `internalId` on each Editor, but it isn't exposed anywhere or passed down to any of the plugins. I'll think on it, perhaps extracting the method or even potentially adding a linting rule to disallow setting ids to prevent these types of errors in the future.

No changes necessary.

Reply:
In this case we need the `id` though. The `Browse` link in the file upload panel is set up as a `label` using the `for` attribute which (as far as I know) has to point to an `id` in order to work. The only alternative would be to nest the upload `input` within the `label` element.

Reply:
Right, to clarify better, the linter wouldn't so much be disallowing setting ids, but rather disallowing setting ids without some sort of validation/sanitization. For instance, I just added the [eslint-plugin-no-unsanitized](https://github.com/mozilla/eslint-plugin-no-unsanitized) which does this exact type of thing for e.g. `innerHTML`.